uniffi_bindgen/src/interface/mod.rs

```diff
@@ -44,27 +44,26 @@
 //!
 //!   * Error messages and general developer experience leave a lot to be desired.
 
-use std::{collections::hash_map::Entry, collections::HashMap, convert::TryFrom, str::FromStr};
+use std::{convert::TryFrom, str::FromStr};
 
 use anyhow::bail;
 use anyhow::Result;
-use serde::{Deserialize, Serialize};
 
 pub mod types;
-use types::TypeResolver;
+use types::TypeUniverse;
 pub use types::{FFIType, Type};
 
 /// The main public interface for this module, representing the complete details of an interface exposed
 /// by a rust component and the details of consuming it via an extern-C FFI layer.
 ///
-#[derive(Debug, Default, Serialize, Deserialize)]
```

Comment:
We no longer attempt to serialize/deserialize `ComponentInterface` structs, so I've removed the derivations here rather than adding them to my new structs.